chainerio/containers/zip.py

```diff
@@ -18,6 +20,20 @@ def __init__(self, base_file_object, base_filesystem_handler,
         if 'b' not in mode:
             base_file_object = io.TextIOWrapper(base_file_object,
                                                 encoding, errors, newline)
+        elif sys.version_info < (3, 7, ):
+            # In the old implemetaion of zipfile before Python 3.7,
+            # the ZipExtFile was not seekable, which makes nested zip
+            # difficult since making a zip requires the file to be seekable.
+            # As a workaround we put the data into BytesIO object.
+
+            warnings.warn('In the current Python, Chainerio has to read '
+                          'the whole file content from the zip '
+                          'on open, which might cause performance or '
+                          'memory issues. '
```

Comment:
I think the message is not clear, and it's hard to estimate the memory usage. How about this?
"In Python 3.6 or older, to open an internal zip file included in a zip file, ChainerIO reads the whole content of the internal zip. If it is large, it may cause memory issue."

Reply:
Since all the files will be read on open, it is not limited to internal zip file case.
How about this: 
"In Python 3.6 or older, to ensure the seekable and readable attribute is correctly set, ChainerIO reads its whole content on open. If the file is large, it may cause performance or memory issue."

Reply:
The scope of "its whole content" is still not clear and hard to estimate the amount of memory needed.

Reply:
"In Python 3.6 or older, ChainerIO reads the whole content of a file on open. It may cause performance or memory issue. For more details, read the chainerio/containers/zip.py:24"

It might be too long to include everything

Reply:
So for example I don't understand opening a file named C (100MB) included in B.zip (1GB) included in A.zip (10GB).  How much memory is needed when that message is printed? Which of A, B, C "a file" means?

Reply:
That is necessary info to use this library confortably.

Reply:
The nested zip case is a little bit confusing as "open_as_container" is an "open" in ChainerIO, so in that case, 
opening C in B.zip in A.zip involves two opens
=> open B.zip
=> open C
According to `reads the whole content of the file to open from zip.`, for me, it means read entire B.zip and then C. Since C is in B.zip and MIGHT already loaded when reading B.zip, so it needs B.zip (1 GB).

For more simple cases, reading C(100MB) in B.zip(1GB), the sentence "reads the whole content of the file to open from zip." means C.

Reply:
"In Python 3.6 or older, ChainerIO reads the whole content of the file to open from zip. Especially, when using open_as_container to open another container in zip, ChainerIO reads that container as well, Such behavior may cause performance or memory issue."

To cover the internal container case